site-manage/site-manage-group-section-role-helper/tool/src/java/org/sakaiproject/site/tool/helper/managegroupsectionrole/impl/SiteManageGroupSectionRoleHandler.java

```diff
@@ -854,268 +870,376 @@ public String processCancelDelete()
-	        			//group.setProviderGroupId(site.getId() + "_" + role);
-	        			group.getProperties().addProperty(group.GROUP_PROP_WSETUP_CREATED, Boolean.TRUE.toString());
-	        			group.getProperties().addProperty(SiteConstants.GROUP_PROP_ROLE_PROVIDERID, role);
-	
-			        	String title = truncateGroupTitle(role);
-			        	group.setTitle(title);
-	        			
-	        			// get the authz group
-	                	String siteReference = siteService.siteReference(site.getId());
-	                	try
-	                	{
-	                		AuthzGroup siteGroup = authzGroupService.getAuthzGroup(siteReference);
-	                		Set<String> usersHasRole = siteGroup.getUsersHasRole(role);
-	                		if (usersHasRole != null)
-	                		{
-	                			for (Iterator<String> uIterator = usersHasRole.iterator(); uIterator.hasNext();)
-	                			{
-	                				String userId = uIterator.next();
-	                				Member member = site.getMember(userId);
-	            					group.addMember(userId, role, member.isActive(), false);
-	                			}
-	                		}
-	                	}
-	                	catch (GroupNotDefinedException e)
-	                	{
-	                		M_log.debug(this + ".processAutoCreateGroup: no authzgroup found for " + siteReference);
-	                	}
-	        		}
-        		}else{
-        			createRandomGroups(roleList, intToSplit);
-        		}
-        	}
-        		
-    		// save the changes
-    		try
-    		{
-    			siteService.save(site);
-    			// reset the form params
-    			resetParams();
-	        } 
-	        catch (IdUnusedException e) {
-	        	M_log.warn(this + ".processAutoCreateGroup: cannot find site " + site.getId(), e);
-	            return null;
-	        } 
-	        catch (PermissionException e) {
-	        	M_log.warn(this + ".processAutoCreateGroup: cannot find site " + site.getId(), e);
-	            return null;
-	        }
-        	
-    	}
+        // reset the warning messages
+        resetTargettedMessageList();
 
-        return "done";
-    }
-    
-    private void createRandomGroups(List<String> roleList, int unit){
-    	String groupTitle = groupSplit ? groupTitleGroup : groupTitleUser;
-    	
-    	if(groupTitle != null && !"".equals(groupTitle)){
-    		//get list of all users:
+        // Check if fields are valid
+        int intToSplit = validateAutoGroupsFields( optionAssign, groupSplit, numToSplitGroup, numToSplitUser, groupTitleGroup, groupTitleUser );
+        int rosterIntToSplit = validateAutoGroupsFields( rosterOptionAssign, rosterGroupSplit, rosterNumToSplitGroup, rosterNumToSplitUser, rosterGroupTitleGroup, rosterGroupTitleUser );
+        if( intToSplit == -1 || rosterIntToSplit == -1 )
+        {
+            return null;
+        }
 
-    		List<String> usersList = new ArrayList<String>();
+        List<String> rosterList = new ArrayList<>();
+        if (!selectedRosters.isEmpty())
```

Comment:
shouldn't need this test for empty, the foreach loop should just skip an empty collection